tests/test_workspace.py

```diff
@@ -721,3 +722,19 @@ def test_sorted(workspace_factory):
     for channel in new_ws['channels']:
         # check sort
         assert channel['samples'][-1]['name'] == 'zzzzlast'
+
+
+def test_closure_over_workspace_build():
+    model = pyhf.simplemodels.hepdata_like(
+        signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0]
+    )
+    data = [51, 48]
+    one = pyhf.infer.hypotest(1.0, data + model.config.auxdata, model)
+
+    workspace = pyhf.workspace.build(model, data)
+    wspace = pyhf.Workspace(json.loads(json.dumps(workspace)))
+    newmodel = wspace.model()
+    newdata = wspace.data(newmodel)
+    two = pyhf.infer.hypotest(1.0, newdata, newmodel)
+
+    assert one == two
```

Comment:
perhaps we can just compute the digest of the two workspaces and assert the equivalence?

Reply:
there is no "previous" workspace,.. this is a method to create a wspace from model + data

Reply:
right, but we should check a "roundtrip" of sorts with simpler workspaces... more complex ones won't be so easy to round-trip.

Reply:
done